packages/flutter/test/material/mergeable_material_test.dart

```diff
@@ -0,0 +1,170 @@
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialGap(),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialGap(),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              )
+            ]
+          )
+        )
+      )
+    );
+
+    await tester.pump(const Duration(milliseconds: 100));
+    expect(box.size.height, lessThan(332));
+
+    await tester.pump(const Duration(milliseconds: 100));
+    expect(box.size.height, equals(332));
+
+    // Remove one gap.
+    await tester.pumpWidget(
+      new Scaffold(
+        body: new ScrollableViewport(
+          child: new MergeableMaterial(
+            key: mmKey,
+            children: <MergeableMaterialItem>[
+              new MaterialSlice(
+                key: sliceKey,
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialGap(),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              )
+            ]
+          )
+        )
+      )
+    );
+
+    await tester.pump(const Duration(milliseconds: 200));
+    expect(box.size.height, equals(316));
+
+    // Add two slices.
+    await tester.pumpWidget(
+      new Scaffold(
+        body: new ScrollableViewport(
+          child: new MergeableMaterial(
+            key: mmKey,
+            children: <MergeableMaterialItem>[
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialSlice(
+                key: sliceKey,
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialGap(),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              ),
+              new MaterialSlice(
+                child: new SizedBox(
+                  width: 100.0,
+                  height: 100.0
+                )
+              )
+            ]
+          )
+        )
+      )
+    );
+
```

Comment:
Right now, the way it works is that it will raise an assertion error in the case where you mess with a slice,row,slice animating block. So A,B,gap,C -> A,gap,B,C will raise. I should probably document the assertion.

To me this behavior makes sense, but it could be changed to be more flexible.